test/int/cluster_test.go

```diff
@@ -68,72 +69,86 @@ var _ = Describe("AtlasCluster", func() {
 		removeControllersAndNamespace()
 	})
 
+	doCommonChecks := func() {
+		Expect(createdCluster.Status.ConnectionStrings).NotTo(BeNil())
+		Expect(createdCluster.Status.ConnectionStrings.Standard).NotTo(BeNil())
+		Expect(createdCluster.Status.ConnectionStrings.StandardSrv).NotTo(BeNil())
+		Expect(createdCluster.Status.MongoDBVersion).NotTo(BeNil())
+		Expect(createdCluster.Status.MongoURIUpdated).NotTo(BeNil())
+		Expect(createdCluster.Status.StateName).To(Equal("IDLE"))
+		Expect(createdCluster.Status.Conditions).To(ConsistOf(testutil.MatchConditions(
+			status.TrueCondition(status.ClusterReadyType),
+			status.TrueCondition(status.ReadyType),
+		)))
+		Expect(createdCluster.Status.ObservedGeneration).To(Equal(createdCluster.Generation))
```

Comment:
can we add one more check - to verify that both fields have increased after the update? This will make sure the updates were not missed by the Operator for any reasons. Some global counter may work out?